elements/lisk-p2p/src/peer/base.ts

```diff
@@ -492,17 +496,28 @@ export class Peer extends EventEmitter {
 			this._productivity = { ...DEFAULT_PRODUCTIVITY };
 		}
 	}
-
 	private _updateFromProtocolPeerInfo(rawPeerInfo: unknown): void {
 		const protocolPeerInfo = {
 			...rawPeerInfo,
 			ip: this._ipAddress,
 			wsPort: this._wsPort,
 		};
+
 		const newPeerInfo = validatePeerInfo(
 			protocolPeerInfo,
 			this._peerConfig.maxPeerInfoSize,
 		);
+
+		if (!this._serverNodeInfo) {
+			throw new Error('Missing server node info.');
+		}
+
+		const result = validatePeerCompatibility(newPeerInfo, this._serverNodeInfo);
+
+		if (!result.success && result.errors) {
+			throw result.errors[0];
```

Comment:
`result.errors[0]` is `string`, so I think it should be `throw new Error(result.errors[0]`?
Also, I don't know why only this function is returning `errors` as array but always one item

Reply:
I'm not sure why but looks like the `validator` function was designed that way. I will change it to single error.